gmso/formats/mol2.py

```diff
@@ -51,165 +51,88 @@ def from_mol2(filename, site_type="atom"):
     return topology
 
 
-def _load_top_sites(f, topology, site_type="atom"):
-    """Take a mol2 file section with the heading '<TRIPOS>ATOM' and save to the topology.sites attribute.
-
-    Parameters
-    ----------
-    f : file pointer
-        pointer file where the mol2 file is stored. The pointer must be at the head of the rti for that
-        `@<TRIPOS>ATOM` section.
-    topology : gmso.Topology
-        topology to save the site information to.
-    site_type : string ('atom' or 'lj'), default='atom'
-        tells the reader to consider the elements saved in the mol2 file, and
-        if the type is 'lj', to not try to identify the element of the site,
-        instead saving the site name.
-
-    Returns
-    -------
-    line : string
-         returns the last line of the `@<TRIPOS>ATOM` section, and this is where the file pointer (`f`)
-         will now point to.
-
-    Notes
-    -----
-    Will modify the topology in place with the relevant site information. Indices will be appended to any
-    current site information.
-
-    """
-    while True:
-        line = f.readline()
-        if _is_end_of_rti(line):
-            line = line.split()
-            position = [float(x) for x in line[2:5]] * u.Å
-            # TODO: make sure charges are also saved as a unyt value
-            # TODO: add validation for element names
-            if site_type == "lj":
-                element = None
-            elif element_by_symbol(line[5]):
-                element = element_by_symbol(line[5])
-            elif element_by_name(line[5]):
-                element = element_by_name(line[5])
-            else:
-                warnings.warn(
-                    "No element detected for site {} with index{}, consider manually adding the element to the topology".format(
-                        line[1], len(topology.sites) + 1
-                    )
-                )
-                element = None
-            try:
-                charge = float(line[8])
-            except IndexError:
-                warnings.warn(
-                    "No charges were detected for site {} with index {}".format(
-                        line[1], line[0]
-                    )
-                )
-                charge = None
-            atom = Atom(
-                name=line[1],
-                position=position.to("nm"),
-                charge=charge,
-                element=element,
-                residue_name=line[7],
-                residue_number=int(line[6]),
+def _parse_atom(top, section, site_type="atom"):
+    """Parse atom information from the mol2 file."""
+    parse_ele = {
+        "lj": lambda ele: None,
+        "atom": lambda ele: element_by_symbol(ele)
+        if element_by_symbol(ele)
+        else element_by_name(ele),
+    }
+    for line in section:
+        cont = line.split()
+        position = [float(x) for x in cont[2:5]] * u.Å
+        element = parse_ele[site_type](cont[5])
```

Comment:
Maybe worthwhile to put a note at the top that mol2 positions are assumed to be in angstroms

Reply:
added to comment